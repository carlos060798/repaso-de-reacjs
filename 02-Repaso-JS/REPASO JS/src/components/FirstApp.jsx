import PropTypes from "prop-types";
// eslint-disable-next-line react/prop-types
function FirstApp({nombre }) {
    // eslint-disable-next-line react/prop-types
    
 
    return ( <>
    <h1>First App</h1>
    <p>{nombre}</p>
    </> );
}

export default FirstApp;

FirstApp.propTypes = {
 nombre: PropTypes.string.isRequired
}