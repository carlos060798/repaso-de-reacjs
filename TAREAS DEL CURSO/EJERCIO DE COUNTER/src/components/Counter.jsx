import { useState } from 'react';
import PropTypes from 'prop-types';

export  default  function Counter ({ value }){
    
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => { 
        // console.log(event)
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 )
    }

    const handleSubstract = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    );
}

Counter.propTypes = {
    value: PropTypes.number.isRequired
}



/*function Counter(props) {

// eslint-disable-next-line react/prop-types
const {valor,handleincrement, handledecrement,Reset} = props
    return ( <>
    <h2>Soy tu contador {valor}</h2> 
  <button onClick={handleincrement}>+1</button>
  <button onClick={handledecrement}>-1</button>
<button onClick={Reset}>Reset</button>
    
    
    </> );
}

export default Counter;*/