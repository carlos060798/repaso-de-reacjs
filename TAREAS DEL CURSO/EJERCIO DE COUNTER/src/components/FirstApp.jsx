import PropTypes from "prop-types";
// eslint-disable-next-line react/prop-types
export function FirstApp({nombre,edad }) {
    // eslint-disable-next-line react/prop-types
    
    return ( <>
      <h1>First App</h1>
    <p data-testid="test-nombre">{nombre}</p>
    <p>{edad}</p>
    </> );
}



FirstApp.propTypes = {
 nombre: PropTypes.string.isRequired,
 edad: PropTypes.number
}

FirstApp.defaultProps = {
    nombre: "Soy un valor por defecto",
    edad: 24
}