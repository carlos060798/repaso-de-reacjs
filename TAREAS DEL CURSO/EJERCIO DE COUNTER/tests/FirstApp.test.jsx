/* 


import {getByTestId, render} from "@testing-library/react";
import { FirstApp } from "../src/components/FirstApp";
// eslint-disable-next-line no-undef
describe("FirstApp", () => {
/* test del snapshot
test("prueba uno hacer macth  snapshot", () => {
 const {container}=render(<FirstApp />);
 expect(container).toMatchSnapshot();
})
test("prueba de mostar nombre con un data-id", () => {
    const nombre = "carlos";
    const {container,getByText,getByTestId}=render(<FirstApp nombre={nombre} />);
    expect(getByText(nombre)).toBeTruthy();
    expect(getByTestId("test-nombre").innerHTML).toBe(nombre);
   })
   

})*/