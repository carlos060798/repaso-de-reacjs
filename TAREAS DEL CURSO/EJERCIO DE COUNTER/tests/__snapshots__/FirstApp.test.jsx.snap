// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp prueba uno hacer macth  snapshot 1`] = `
<div>
  <h1>
    First App
  </h1>
  <p>
    Soy un valor por defecto
  </p>
  <p>
    24
  </p>
</div>
`;
